crates/compiler/roar/src/proc.rs

```diff
@@ -0,0 +1,30 @@
+use crate::ops::{Operation};
+use crate::storage::{Register,FloatRegister,Constant,ByteSize,Offset,Input,Output};
+use roc_module::symbol;
+#[derive(Clone,Debug)]
+#[non_exhaustive]
+pub struct Proc {
+    ///The instructions themselves
+    pub instructions : Vec<Operation>,
+    ///A list of inputs, in order, taken by the function
+    ///The reason it's type is `Output` and not `Input` is that we want the values to be written from certain registers, not read from them
+    pub inputs : Vec<Output>,
+    ///A list of computed registers used by the operation
+    pub annotations : Annotations
+}
+///Some annotations on the internal level that may or may not be present
+#[derive(Clone,Debug)]
+pub struct Annotations {
+    ///All the registers it uses
+    uses : Option<Vec<Output>>,
+    ///It's name in the AST 
+    symbol : Option<symbol::Symbol>,
```

Comment:
I don't really follow these and when they would or would not be used (like a proc always has a symbol, right?), but I assume you have a goal for them.

Reply:
For symbol, it is for the purpose of debugging (if the program has a runtime error), and as for the `uses`, it is a list of registers modified by the procedure, which is chached (for a given function) after it's first computation